File: BinarySearchArrayList.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package exercises;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;

/**
 * @BingChat
 * @author MOPHE
 */
public class BinarySearchArrayList {

    public static void main(String[] args) {
        String path = "C:\\Users\\MOPHE\\Documents\\NetBeansProjects\\Exercises\\src\\exercises\\files\\english.txt";
        ArrayList<String> list = new ArrayList<>();
        try {
            File file = new File(path);
            try (Scanner scanner = new Scanner(file)) {
                scanner.useDelimiter("\n");
                while (scanner.hasNextLine()) {
                    String line = scanner.nextLine();
                    list.add(line);
                }
            }
        } catch (FileNotFoundException e) {
            System.out.println("An error occurred.");
        }
        Collections.sort(list);
        System.out.println(list);

        Scanner sc = new Scanner(System.in);
        String input = sc.next();

        int index = Collections.binarySearch(list, input);
        System.out.println(input + " is found at index " + index);
    }
}

File: binarySearchArrayList.py
import bisect

# with open('Files/english.txt', 'r') as f:
#     lines = f.readlines()
#     lines = [line.strip() for line in lines]
#     lines.sort()
#
# print(lines)

with open('Files/english.txt', encoding='utf-8') as file:
    l = [line.strip() for line in file]

l = sorted(l)
print(l)

item = input("What do you want to search? ")
index = bisect.bisect_left(l, item)

print(f"{item} exists at index {index}")
